### PR TITLE
start of coverage for vector instructions  (for coverage)(2nd pull attempt)

### PR DESCRIPTION
**Pull request by simond-imperas**
adding vector log to csv for ovpsim - added to csv file table and tiiedup ovpsim log file processing

### CODE REVIEW

File: scripts/ovpsim_log_to_trace_csv.py [hunk 1]
```diff
@@ -17,14 +18,28 @@
 """
 import re
 import os
+#import os.path
```
Comment (taoliug) [inline scripts/ovpsim_log_to_trace_csv.py]: Can you remove all commented code, or add TODO if plan to enable in the future?

File: scripts/ovpsim_log_to_trace_csv.py [hunk 2]
```diff
@@ -17,14 +18,28 @@
...
+try:
+    from ovpsim_log_to_trace_csv_vectors import *
+except:
+    def assign_operand_vector(a,b,c,d):
+        logging.fatal("No OVPsim vector instruction processing included")
+        sys.exit(-1)
+
+stop_on_first_error = 0
+
+def fatal (s):
```
Comment (taoliug) [inline scripts/ovpsim_log_to_trace_csv.py]: Add docstring for every function, here and below

File: scripts/ovpsim_log_to_trace_csv.py [hunk 3]
```diff
@@ -96,7 +120,8 @@ def process_if_compressed(prev_trace):
     'j'         :'jal',
     'jr'        :'jal',
     }
-def check_conversion(entry, stop_on_first_error):
+
+def check_conversion(entry ):
```
Comment (taoliug) [inline scripts/ovpsim_log_to_trace_csv.py]: Remove space after "entry"

File: scripts/riscv_trace_csv.py [hunk 4]
```diff
@@ -89,7 +106,23 @@ def write_trace_entry(self, entry):
                               'instr'   : entry.instr,
                               'imm'     : entry.imm,
                               'csr'     : entry.csr,
-                              'binary'  : entry.binary})
+                              'binary'  : entry.binary,
+                              'mode'    : entry.privileged_mode,
+                              'vd'      : entry.vd,
+                              'vd_val'  : entry.vd_val,
+                              'vs1'      : entry.vs1,
```
Comment (taoliug) [inline scripts/riscv_trace_csv.py]: Alignment for the colon, here and below

File: scripts/ovpsim_log_to_trace_csv.py [hunk 5]
```diff
@@ -248,12 +326,18 @@ def main():
                                          help="Generate the full trace")
   parser.add_argument("-v", "--verbose", dest="verbose", action="store_true",
                                          help="Verbose logging")
+  parser.add_argument("--stop_on_first_error", dest="stop_on_first_error", action="store_true",
```
Comment (taoliug) [inline scripts/ovpsim_log_to_trace_csv.py]: Fix line width, here and below.

File: scripts/ovpsim_log_to_trace_csv.py [hunk 6]
```diff
@@ -115,129 +140,182 @@ def check_conversion(entry, stop_on_first_error):
...
-                sys.exit(-1)
+        else: 
+            line = line.strip()
+            if verbose2: logging.debug("ignoring line: [%s] %s " % (str(instr_cnt), line))
+            line = re.sub(' +', ' ', line)
+            split = line.split(" ")
+            if len(split) == 1: continue
+            item = split[1]
+            if "----" in item: continue
+            if "REPORT" in line or item in ["mtvec","pmpaddr0","pmpcfg0","mstatus","mepc","mscratch","mcause","mtval","vl","vtype"]:
```
Comment (taoliug) [inline scripts/ovpsim_log_to_trace_csv.py]: Line length should be <= 80 characters, please fix here and elsewhere.

**Comment by simond-imperas** [comment]
superceded

**Comment by taoliug** [comment]
May i know whey this PR is closed? Can we keep iterating on this PR rather than opening a new one?

**Comment by simon5656** [comment]
sorry as a novice git user - I must have done this wrong... I will learn more for future

________________________________

**Comment by taoliug** [comment]
No worry, will review your new PR.

**Pull request closed without merging**